guardian/shortcuts.py

```diff
@@ -3,21 +3,68 @@
 """
 from __future__ import unicode_literals
 from django.apps import apps
-from django.contrib.auth.models import Group, Permission
+from django.contrib.auth.models import Group
+from django.contrib.auth.models import Permission
 from django.contrib.contenttypes.models import ContentType
-from django.db.models import Count, Q, QuerySet
+from django.db.models import Count
+from django.db.models import Q
+from django.db.models import QuerySet
 from django.shortcuts import _get_queryset
-from guardian.compat import basestring, get_user_model
+from guardian.compat import basestring
+from guardian.compat import get_user_model
 from guardian.core import ObjectPermissionChecker
 from guardian.ctypes import get_content_type
-from guardian.exceptions import MixedContentTypeError, WrongAppError
-from guardian.utils import get_anonymous_user, get_group_obj_perms_model, get_identity, get_user_obj_perms_model
+from guardian.exceptions import MixedContentTypeError
+from guardian.exceptions import WrongAppError
+from guardian.utils import get_anonymous_user
+from guardian.utils import get_group_obj_perms_model
+from guardian.utils import get_identity
+from guardian.utils import get_user_obj_perms_model
 from itertools import groupby
 
 import warnings
 
 
-def assign_perm(perm, user_or_group, obj=None):
+def to_permission(perm):
+    if not isinstance(perm, Permission):
+        try:
+            app_label, codename = perm.split('.', 1)
+        except ValueError:
+            raise ValueError("For global permissions, first argument must be in"
+                             " format: 'app_label.codename' (is %r)" % perm)
+        perm = Permission.objects.get(content_type__app_label=app_label,
+                                      codename=codename)
+    return perm
+
+
+def assign_perm_from_user_group_object(perm, user_group_object):
+    ''' Assigns permission to the user and content_object in the
+        UserGroupObject.
+    '''
+    perm = to_permission(perm)
+    model = get_user_obj_perms_model(user_group_object.content_object)
+    return model.objects.assign_perm_from_user_group_object(perm, user_group_object)
+
+
+def assign_perm_from_user_group_objects(perm, user_group_objects):
+    ''' Assigns permission to the user and content_objects in the
```

Comment:
Turns out that I had already gone back and modified the bulk_assign_perm_from_user_group_objects to be able to accept heterogeneous users. I'll update the documentation.